framework/include/utils/RankTwoTensor.h

```diff
@@ -111,37 +116,124 @@ class RankTwoTensorTempl : public TensorValue<T>
     general = 9
   };
 
-  // Deprecated constructor (replaced by initializeFromRows)
-  RankTwoTensorTempl(const TypeVector<T> & row1,
-                     const TypeVector<T> & row2,
-                     const TypeVector<T> & row3);
+  /**
+   * @brief Initialize the random seed based on an unsigned integer.
+   *
+   * Deprecated in favor of MooseRandom::seed().
+   */
+  static void initRandom(unsigned int);
 
   /**
-   * Named constructor for initializing symetrically. The supplied vectors are
-   * used as row and column vectors to construct two tensors respectively, that
-   * are averaged to create a symmetric tensor.
+   * @brief Get the available `FillMethod` options.
+   *
+   * This method is useful in validParams().
    */
-  [[nodiscard]] static RankTwoTensorTempl
-  initializeSymmetric(const TypeVector<T> & v0, const TypeVector<T> & v1, const TypeVector<T> & v2);
+  [[nodiscard]] static MooseEnum fillMethodEnum();
 
-  /// Named constructor for initializing from row vectors
-  [[nodiscard]] static RankTwoTensorTempl initializeFromRows(const TypeVector<T> & row0,
-                                                             const TypeVector<T> & row1,
-                                                             const TypeVector<T> & row2);
+  /**
+   * @brief Fill a `libMesh::TensorValue<T>` from this second order tensor
+   */
+  void fillRealTensor(libMesh::TensorValue<T> &);
 
-  /// Named constructor for initializing from column vectors
-  [[nodiscard]] static RankTwoTensorTempl initializeFromColumns(const TypeVector<T> & col0,
-                                                                const TypeVector<T> & col1,
-                                                                const TypeVector<T> & col2);
+  /// Print the rank two tensor
+  void print(std::ostream & stm = Moose::out) const;
+
+  /// Print the Real part of the RankTwoTensorTempl<ADReal>
+  void printReal(std::ostream & stm = Moose::out) const;
+
+  /// Print the Real part of the RankTwoTensorTempl<ADReal> along with its first nDual dual numbers
+  void printDualReal(unsigned int nDual, std::ostream & stm = Moose::out) const;
+
+  /// @}
+
+  /// @{
+
+  /**
+   * @brief Empty constructor; fills to zero
+   *
+   * \f$ A_{ij} = 0 \f$
+   *
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{.cpp}
+   * RankTwoTensor A;
+   * // A = [ 0 0 0
+   * //       0 0 0
+   * //       0 0 0 ]
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   */
+  RankTwoTensorTempl();
 
-  /// Constructor that proxies the fillFromInputVector method
+  /**
+   * @brief Select specific initialization pattern.
+   *
+   * `initNone` initializes an empty second order tensor.
+   *
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{.cpp}
+   * RankTwoTensor A(RankTwoTensor::initNone);
+   * // A = [ 0 0 0
+   * //       0 0 0
+   * //       0 0 0 ]
```

Comment:
`initNone` should leave the data _uninitialized_. It is to be used whenever you are setting the tensor entries to custom values rightaway.

Reply:
Fixed